submissions/Harnytskyi/js-post-oop/index.js

```diff
@@ -0,0 +1,104 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://harnytskyi.github.io/a-tiny-JS-world
+   Web app: https://github.com/Harnytskyi/a-tiny-JS-world
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+class Inhabitant {
+    constructor(species, name, gender, saying, friends = []) {
+        this.species = species;
+        this.name = name;
+        this.gender = gender;
+        this.saying = saying;
+        this.friends = friends;
+    };
+    getListFriends(){
+        if (this.friends.length == 0){
+            return "I have not a friends."
+        }
+        else{
+            var listFriends = [];
+            this.friends.map(element => {
+                listFriends.push(element.name);
+            })
```

Comment:
This way of using `Array#map` is exactly the way to use `Array#forEach`.
The beauty of `Array#map` is that it as able to transform every item in inbound array. It will make code shorter and better readable.

Check the following materials to master this powerful method. It will come to a rescue in many cases.

- https://dev.to/sarah_chima/the-javascript-array-map-method-44m0
- https://www.javascripttutorial.net/javascript-array-map/